.github/workflows/tlaplus.yml

```diff
@@ -57,3 +57,72 @@ jobs:
         with:
           name: tlc
           path: tla/*.out
+
+  model-checking-consistency:
+    name: Model Checking (Consistency)
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v3
+      - run: python ./tla/install_deps.py
+
+      - name: consistency/SingleNode.cfg
+        run: |
+          set -exo pipefail
+          cd tla/
+          ./tlc.sh -config consistency/SingleNode.cfg consistency/SingleNode.tla 2>&1 | tee consistency/SingleNode.out
```

Comment:
Minor: It should be possible to reduce the copy-pasting here with an [`each`](https://learn.microsoft.com/en-us/azure/devops/pipelines/process/expressions?view=azure-devops#each-keyword). Something like:

```
# At top of file
parameters:
- name: consistencyFiles
  default:
  - consistency/SingleNode
  - consistency/SingleNodeReads
  ...

# Here
- ${{ each value in parameters.consistency}}:
  - name ${{ value }}
  - run: |
      set -exo pipefail
      cd tla/
      ./tlc.sh -config ${{ value }}.cfg ${{ value }}.tla 2>&1 | tee ${{ value }}.out
```

However, YAML (and ADO's YAML templating in-particular) are evil _and_ unhelpful, so getting the syntax correct is an annoying trial of errors.

Reply:
Having done that, a lot, on the main CI, I think I can safely say that I don't want to do it for just three values on one dimension.

Reply:
I'm suggesting using this as a simple template for the tasks below, so 9 (or 27) repetitions.

Reply:
fwiw, this a sample in GHA using four dimensions https://github.com/microsoft/CCF/blob/main/.github/workflows/containers.yml

Reply:
The matrix thing is nice mostly if you have A x B x C with only a few exceptions, in which case it will be substantially less work to tweak the dimensions than with the current denormalised scheme. If combinations are sparse and ad-hoc, then it's not a particularly good idea to use it.